### DISCUSSION TITLE
Nvim cmp completions not working

### DISCUSSION THREAD

**Discussion by PriceHiller** [nvim-neorg/neorg#352] [Q&A] [answered]
Hey, great project y'all have, unfortunately I'm facing some issues with completions. I've gone through a bunch of issues and discussions here as well as the wiki and I am still unable to get `Nvim-Cmp` to work with `Neorg`.

Here are my relevant configs:
- [Nvim CMP Config](https://gitlab.orion-technologies.io/philler/dot-files/-/blob/Development/.config/nvim/lua/plugins/configs/_cmp.lua)
- [Neorg Config](https://gitlab.orion-technologies.io/philler/dot-files/-/blob/Development/.config/nvim/lua/plugins/configs/_neorg.lua)
- [Treesitter Config](https://gitlab.orion-technologies.io/philler/dot-files/-/blob/Development/.config/nvim/lua/plugins/configs/treesitter.lua)
- [Plugins](https://gitlab.orion-technologies.io/philler/dot-files/-/blob/Development/.config/nvim/lua/plugins/plugins.lua)

Here's my `Neorg` config without having to go through a link:

```lua
local neorg = require("neorg")

neorg.setup({
    load = {
        ["core.defaults"] = {},
        ["core.norg.concealer"] = {},
        ["core.norg.esupports.metagen"] = {
          config = {
              type = "auto",
          }
        },
        ["core.norg.completion"] = {
            config = {
                engine = "nvim-cmp"
            },
        },
        ["core.keybinds"] = {
            config = {
                default_keybinds = true,
                norg_leader = "<Leader>o"
            }
        },
        ["core.norg.dirman"] = {
            config = {
                workspaces = {
                    default = "~/.neorg"
                }
            }
        },
        ["core.integrations.telescope"] = {},
    }
})
```

I'm at a bit of a loss as to why I'm not getting completions, as I *seemingly* have my `Neorg` configuration correct and I have packer set to load `Neorg` *after* `Treesitter`.

When I have a `.norg` file open and run `CmpStatus` I can see that the `neorg` source *is* popping up, but showing as unused:
```
# ready source names
- buffer
- emoji
- path

# unused source names
- neorg
- cmdline

# unknown source names
- luasnip
- nvim_lsp
- nvim_lua
```
I attempted to add `Neorg` as a source to `Cmp`, but that did not provide completions.

I also had the option:
```lua
        ["core.integrations.nvim-cmp"] = {
            config = {}
        }
```
set in there previously instead of `core.norg.completion` and with `core.norg.completion`, but that did not yield any success either.

If there's anything that stands out I'd much appreciate feedback on what I have incorrect, thanks.

**Comment by PriceHiller** [comment, marked as answer]
Nevermind, I've resolved this mess... ended up burning my ~/.local/share/nvim directory down and reinstalling packer from bootstrap and now it works flawlessly... *very* annoying. Apologies for filling up the discussions.

**Reply by rawsp33d** [reply]
Do you have any more insight about how that fixed it? I have the same problem and tried the same fix to no avail.

**Reply by rawsp33d** [reply]
Ah, in my case I hadn't added `neorg` as a completion source to cmp. Apologies for filling up the comments to the discussion that filled up the discussions.
